src/components/characterList.jsx

```diff
@@ -3,18 +3,14 @@ import Character from './Character';
 
 const CharacterList = ({ scriptOption }) => {
   const [characters, setCharacters] = useState([]);
-  const fetchedCharacters = [];
 
   // get the character data on the initial load and when the script option changes
   useEffect(() => {
     // fetch request to get the character data
     fetch('/script/' + scriptOption + '/characters')
       .then((response) => response.json())
-      .then((data) => {
-        for (let name in data.characterData) {
-          fetchedCharacters.push(data.characterData[name]);
-        }
-        setCharacters(fetchedCharacters);
+      .then((characterData) => {
+        setCharacters(characterData);
```

Comment:
This might be _less_ readable, but I think we could just pass the setCharacters function directly since it has the same parameter as the arrow function.
```js
.then(setCharacters)
```